http-client/src/main/java/io/micronaut/http/client/interceptor/HttpClientIntroductionAdvice.java

```diff
@@ -356,9 +359,24 @@ public Object intercept(MethodInvocationContext<Object, Object> context) {
                                         request, publisherArgument
                                 );
                             } else {
-                                publisher = streamingHttpClient.dataStream(
+                                Publisher<ByteBuffer<?>> byteBufferPublisher = streamingHttpClient.dataStream(
                                         request
                                 );
+                                if (argumentType == ByteBuffer.class) {
+                                    publisher = byteBufferPublisher;
+                                } else if (argumentType == byte[].class) {
+                                    publisher = Flowable.fromPublisher(byteBufferPublisher)
+                                            .map(buffer -> {
+                                                byte[] bytes = buffer.toByteArray();
+                                                ((ReferenceCounted) buffer).release();
+                                                return bytes;
+                                            });
+                                } else {
+                                    throw new ConfigurationException("Cannot create the generated HTTP client's " +
```

Comment:
I think it would be better to fall back to a Flowable that tries to `ConversionService.SHARED.convert(buffer, argument)` and fails only if the conversion is not possible.

Reply:
That would be more general, but could lead to many surprises when network buffer or HTTP chunk boundaries split the byte stream in unexpected places. Save for byte[] and perhaps NIO ByteBuffers, what would be a relevant conversion?

Reply:
@jespersm The idea is that the user could decide to register a type converter if they wished. If you have an argument type of a byte[], then no conversion will take place since the types will match.

Reply:
I understand the mechanics, but not the benefit. I can only imagine people trying to register converters from ByteBuffer to String, which would work for tiny test cases, but fail in real life.

Reply:
Or, like in the test case: If you register a converter for ByteBuffer to Boolean, it would likely work in the test case, but it would depend on timing and packet sizes (Nagle's algorithm). We shouldn't rely on higher leven data structures without proper framing, and this will only invite users to do so.

Reply:
So yes, I've updated the PR, but I still think it's bad idea.

Also, too bad we need to look up the TypeConverter for every packet, because io.micronaut.core.convert.DefaultConversionService#findTypeConverter is not interface.